src/autocomplete.js

```diff
@@ -165,7 +164,7 @@ export default class Autocomplete extends Component {
       menuOpen: newState.menuOpen || false,
       query: newQuery,
       selected: null
-    })
+    }, this.props.confirmOnBlur ? this.props.onConfirm(newQuery) : null)
```

Comment:
Thanks @tvararu, I will correct the callback application, but as far as sending `options[selected]` to `onConfirm`, wouldn't that be misleading? I thought `onConfirm` is meant to receive the selected query, in which case `options[-1]` was giving `undefined`.

What I mean is, in normal usage the `selected` property was not being updated in the `state` in all cases or would come back as `-1`. I was just using any of the example Autocompletes.